### PR TITLE
Rewrite if-else to switch statements

### PR DESCRIPTION
**Pull request by sosiska**
This commit rewrites some if-else-if-else chains as a switch.
Based on Go style guide:
https://golang.org/doc/effective_go.html#switch

Signed-off-by: Kirill Motkov <motkov.kirill@gmail.com>

### CODE REVIEW

**Comment by black-adder** [comment]
@sosiska thanks for the PR, do you mind signing off on all your commits? Instructions can be found here: https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md#certificate-of-origin---sign-your-work

**Comment by sosiska** [comment]
> @sosiska thanks for the PR, do you mind signing off on all your commits? Instructions can be found here: https://github.com/jaegertracing/jaeger/blob/master/CONTRIBUTING_GUIDELINES.md#certificate-of-origin---sign-your-work

As I can see, my commit 2ef240bdaaa162d5f8df3ed08ef10d17131dc8a2 is already signed.
```Signed-off-by: Kirill Motkov <motkov.kirill@gmail.com>```

**Comment by yurishkuro** [comment]
Commit sha: 2ef240b, Author: Motkov Kirill, Committer: Motkov Kirill; Expected "Motkov Kirill motkov.kirill@gmail.com", but got "Kirill Motkov motkov.kirill@gmail.com".

DCO bot is picky about the exact match of signature / git config / github account.

File: cmd/collector/app/sanitizer/zipkin/span_sanitizer.go [hunk 1]
```diff
@@ -168,11 +168,12 @@ func (s *errorTagSanitizer) Sanitize(span *zc.Span) *zc.Span {
 		if binAnno.AnnotationType != zc.AnnotationType_BOOL && strings.EqualFold("error", binAnno.Key) {
 			binAnno.AnnotationType = zc.AnnotationType_BOOL
 
-			if strings.EqualFold("true", string(binAnno.Value)) || len(binAnno.Value) == 0 {
+			switch {
+			case strings.EqualFold("true", string(binAnno.Value)), len(binAnno.Value) == 0:
```
Comment (yurishkuro) [inline cmd/collector/app/sanitizer/zipkin/span_sanitizer.go]: I find this much harder to read (especially the fact that this is an OR

Comment (sosiska) [reply cmd/collector/app/sanitizer/zipkin/span_sanitizer.go]: Return if-else or replace `,` with a `||`?

Comment (yurishkuro) [reply cmd/collector/app/sanitizer/zipkin/span_sanitizer.go]: if you keep `||` then fine to do a switch. BTW, better to move `len(binAnno.Value) == 0` in the first position (left of `||`)

Comment (sosiska) [reply cmd/collector/app/sanitizer/zipkin/span_sanitizer.go]: Done in 20cd8d933f8eea211db522bfed1c4cf2a30e55a0.

**Pull request merged**